File: problem-set-1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int cents;

    // Fetch the number of cents until the user enters a value >= 0
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    int coins = 0;

    // Number of coins 25¢
    coins += cents / 25;
    cents %= 25;

    // Number of coins 10¢
    coins += cents / 10;
    cents %= 10;

    // Number of coins 5¢
    coins += cents / 5;
    cents %= 5;

    // Number of coins 1¢
    coins += cents;

    printf("%i\n", coins);
};

File: problem-set-6/sentimental-cash/cash.py
from cs50 import get_float;

# Ask for a value until the user provides a non-negative number
while True:
    change = get_float("Change owed: ")
    if change >= 0:
        break

# Converting dollars to cents (we round up to avoid float problems)
cents = round(change * 100)

coins = 0

# We use the largest coins sequentially: 25, 10, 5, 1
for coin_value in [25, 10, 5, 1]:
    coins += cents // coin_value
    cents %= coin_value

print(coins)
